i18n(fr): Translate list.mdx, notification.mdx, prerequisites/index.mdx, upgrade-migrate/from-tauri-1.mdx (#1445)
tauri-apps/tauri-docs: src/content/docs/fr/2/guide/upgrade-migrate/from-tauri-1.mdx

---

> title: Mise à niveau depuis Tauri 1.0

Reviewer: ```suggestion
title: Mise à Niveau Depuis Tauri 1.0

```

Title case

i18nReady: true
---

import { Tabs, TabItem } from '@astrojs/starlight/components';
import CommandTabs from '@components/CommandTabs.astro';

Ce guide vous explique comment mettre à niveau votre application basée sur Tauri 1.0 vers Tauri 2.0.

> ## Migration automatisée

Reviewer: ```suggestion
## Migration Automatisée

```

Tauri v2 contient une commande `migrate` qui automatise la plupart des processus et vous aide à finir votre migration:

<CommandTabs
	npm="npm install @tauri-apps/cli@latest
    npm run tauri migrate"
	yarn="yarn upgrade @tauri-apps/cli --latest
    yarn tauri migrate"
	pnpm="pnpm update @tauri-apps/cli --latest
    pnpm tauri migrate"
	cargo='cargo install tauri-cli version "^2.0.0-alpha"
    cargo tauri migrate'
/>

{/* TODO: 2.0 */}

Apprenez-en plus à-propos de la commande `migrate` dans l'[interface de référence des commandes](/2/reference/cli#migrate) 
Learn more about the `migrate` command in the [Command Line Interface reference](/2/reference/cli#migrate)

> ## Sommaire des changements

Reviewer: ```suggestion
## Sommaire des Changements

```

- `@tauri-apps/api/http` module enlevé. Utilisez `@tauri-apps/plugin-http` à la place. [Migration](#migrer-vers-un-plugin-http)
- `@tauri-apps/api/os` module enlevé. Utilisez `@tauri-apps/plugin-os` à la place. [Migration](#migrer-vers-un-os-plugin)
- `@tauri-apps/api/notification` module enlevé. Utilisez `@tauri-apps/plugin-notification` à la place. [Migration](#migrer-vers-un-notification-plugin)
- `@tauri-apps/api/process` module enlevé. Utilisez `@tauri-apps/plugin-process` à la place. [Migration](#migrer-vers-un-process-plugin)
- `@tauri-apps/api/shell` module enlevé. Utilisez `@tauri-apps/plugin-shell` à la place. [Migration](#migrer-vers-un-shell-plugin)
- `@tauri-apps/api/updater` module enlevé. Utilisez `@tauri-apps/plugin-updater` à la place [Migration](#migrer-vers-un-updater-plugin)
- `@tauri-apps/api/window` module enlevé. Utilisez `@tauri-apps/plugin-window` à la place [Migration](#migrer-vers-un-window-plugin)

## Etapes de Migration Détaillées

Les scénarios communs que vous pouvez rencontrer lors de votre migration de Tauri 1.0 vers Tauri 2.0.

> ### Migrer votre "App Plugin"

Reviewer: Should "App Plugin" be in quotes here I wonder? Similar for all following headings

Author: I put some quotes here because i can't translate them without loosing the meaning, in this way the reader is not confuse to find some non-French in the sentence. But again, the reader will understand, even more because he probably know why he is here for.

Reviewer: Is there a translation for "Plugin" that would work? Because then you could do `Merger votre "App" [French word for plugin]`

Reviewer: And the same would apply to the additional headings below.

Author: This is the problem, there is no French equivalent for "plugin" his definition, that's why everybody who need to talk about it says "plugin" but it is not in the french dictionnary. Your solution makes sens, `Migrer votre "App" plugin` is understandable. However since you're out the quotes, you need to apply the french rules and invert the words and put `de` between which would makes : `Migrer votre plugin d'"App"` and actually looses the meaning at the same time. If it still understandable for this one, some like `Migrer vers un plugin de "Clipboard"` or even translate clipboard like `Migrer vers un plugin de presse-papier` is unthinkable because it really makes no sens.

Reviewer: I'll leave it to you since you're much more the expert, but I think my preference would be "Migrer votre App Plugin" and to think of "App Plugin" as a proper noun in a way (especially since the package names are `@tauri-apps/plugin-app` for example).

> les APIs JavaScript `@tauri-apps/api/app` ont été enlevées. Utilisez le plugin `@tauri-apps/plugin-app` à la place:

Reviewer: ```suggestion
Les APIs JavaScript `@tauri-apps/api/app` ont été enlevées. Utilisez le plugin `@tauri-apps/plugin-app` à la place:

```

---
title: Mise à Niveau Depuis Tauri 1.0

i18nReady: true
---

import { Tabs, TabItem } from '@astrojs/starlight/components';
import CommandTabs from '@components/CommandTabs.astro';

Ce guide vous explique comment mettre à niveau votre application basée sur Tauri 1.0 vers Tauri 2.0.

## Migration Automatisée

> Tauri v2 contient une commande `migrate` qui automatise la plupart des processus et vous aide à finir votre migration:

Reviewer 2: I think I would keep it simple here

```suggestion
Tauri v2 contient la commande `migrate` qui simplifie votre migration:
```

Author: Why not, I just followed the wording of the english documentation

<CommandTabs
	npm="npm install @tauri-apps/cli@latest
    npm run tauri migrate"
	yarn="yarn upgrade @tauri-apps/cli --latest
    yarn tauri migrate"
	pnpm="pnpm update @tauri-apps/cli --latest
    pnpm tauri migrate"
	cargo='cargo install tauri-cli version "^2.0.0-alpha"
    cargo tauri migrate'
/>

{/* TODO: 2.0 */}

> Apprenez-en plus à-propos de la commande `migrate` dans l'[interface de référence des commandes](/2/reference/cli#migrate) 

Reviewer 2: Not sure about `interface de référence des commandes`

Would not be easier to say `guide des commandes` ou `manuel des commandes`

Author: I wasn't able to remember me the word during the translation, but i think I have even better : what about "Répertoire" ?

> Apprenez-en plus à-propos de la commande `migrate` dans l'[interface de référence des commandes](/2/reference/cli#migrate) 

Reviewer: ```suggestion
Apprenez-en plus à-propos de la commande `migrate` dans l'[interface de référence des commandes](/fr/2/reference/cli#migrate) 

```

> Learn more about the `migrate` command in the [Command Line Interface reference](/2/reference/cli#migrate)

Reviewer: ```suggestion
Learn more about the `migrate` command in the [Command Line Interface reference](/fr/2/reference/cli#migrate)

```

## Sommaire des Changements

Voici ci-dessous un sommaire des changement entre Tauri 1.0 et Tauri 2.0:

### Configuration Tauri

- `tauri > allowlist` retiré.
- `tauri > allowlist > protocol > assetScope` déplacé vers `tauri > security > assetProtocol > scope`.
- `tauri > cli` déplacé vers `plugins > cli`.
- `tauri > updater` déplacé vers `tauri > bundle > updater`.
- `tauri > updater > dialog` retiré.
- `tauri > updater > endpoints` déplacé vers `plugins > updater`.
- `tauri > systemTray` renomé en `tauri > trayIcon`.

> [Référence de l'API de configuration de Tauri 2.0](/2/reference/config)

Reviewer: ```suggestion
[Référence de l'API de configuration de Tauri 2.0](/fr/2/reference/config)

```

### Nouvelles Fonctionnalités de Cargo

- linux-protocol-body: Permet les protocols de requêtes "body parsing", autorisant l'IPC à l'utiliser. Nécéssite webkit2gtk 2.40.

> ### Fonctionnalités de Cargo Retirées

Reviewer 2: Not sure about this one tbh

```suggestion
### Fonctionnalités Cargo Retirées
```

Author: Both are working, I guess yours is more concise

Les APIs Rust `tauri::api::http` et JavaScript `@tauri-apps/api/http` ont été retirées. Utilisez le plugin `@tauri-apps/plugin-http` à la place:

1. Ajoutez vos dépendances Cargo:

```toml
# Cargo.toml
[dependencies]
tauri-plugin-http = "2"
```

2. Utilisez-les dans un projet JavaScript ou Rust:

<Tabs>
<TabItem label="JavaScript">

```rust
fn main() {
    tauri::Builder::default()
        .plugin(tauri_plugin_http::init())
}
```

```json
// package.json
{
	"dependencies": {
		"@tauri-apps/plugin-http": "^2.0.0"
	}
}
```

```js
import { fetch } from '@tauri-apps/plugin-http';
const response = await fetch(
	'https://raw.githubusercontent.com/tauri-apps/tauri/dev/package.json'
);
```

</TabItem>
<TabItem label="Rust">

```rust
use tauri_plugin_dialog::reqwest;

tauri::Builder::default()
    .plugin(tauri_plugin_http::init())
    .setup(|app| {
        let response_data = tauri::async_runtime::block_on(async {
            let response = reqwest::get(
                "https://raw.githubusercontent.com/tauri-apps/tauri/dev/package.json",
            )
            .await
            .unwrap();
            response.text().await
        })?;
        Ok(())
    })
```

> Le plugin HTTP est "re-exports" (ré-exporté) [reqwest](https://docs.rs/reqwest/latest/reqwest/) pour que vous puissiez avoir accès à la documentation pour plus d'informations.

Reviewer 2: ```suggestion
Le plugin HTTP "re-exports" (expose) [reqwest](https://docs.rs/reqwest/latest/reqwest/) pour que vous puissiez avoir accès à la documentation pour plus d'informations.
```

Author: my bad

		"@tauri-apps/plugin-notification": "^2.0.0"
	}
}
```

```js
import { sendNotification } from '@tauri-apps/plugin-notification';
sendNotification('Tauri is awesome!');
```

</TabItem>
<TabItem label="Rust">

```rust
use tauri_plugin_notification::{NotificationExt, PermissionState};

fn main() {
    tauri::Builder::default()
        .plugin(tauri_plugin_notification::init())
        .setup(|app| {
            if app.notification().permission_state()? == PermissionState::Unknown {
                app.notification().request_permission()?;
            }
            if app.notification().permission_state()? == PermissionState::Granted {
                app.notification()
                    .builder()
                    .body("Tauri is awesome!")
                    .show()?;
            }
            Ok(())
        })
}
```

</TabItem>
</Tabs>

### Migrer vers un "Menu Module"

Les APIs Rust `Menu` ont été déplacées vers le module `tauri::menu` et remaniée pour utiliser la [crate muda](https://github.com/tauri-apps/muda).

L'API JavaScript `@tauri-apps/api/window` `onMenuClicked()` va être migrée dans une prochaine version.

#### Utilisez `tauri::menu::MenuBuilder`

> Utilisez `tauri::menu::MenuBuilder` à la place de `tauri::Menu`. Notez que sont constructeur prend en argumen une instance de "Manager" (parmis `App`, `AppHandle` ou `Window`):

Reviewer 2: ```suggestion
Utilisez `tauri::menu::MenuBuilder` à la place de `tauri::Menu`. Notez que son constructeur prend en argument une instance de "Manager" (parmis `App`, `AppHandle` ou `Window`):
```

Author: sorry about that, It was little late when I did the second half of this file

        Ok(())
    })
```

`tauri::Builder::menu` prend maintenant une "closure" car le menu a besoin d'une instance de "Manager" pour être construit. Voir [la documentation](https://docs.rs/tauri/2.0.0-alpha/tauri/struct.Builder.html#method.menu) pour plus d'information.

#### Evenement du Menu

L'API Rust `tauri::Builder::on_menu_event` a été retirée. Utilisez `tauri::App::on_menu_event` ou `tauri::AppHandle::on_menu_event` à la place:

```rust
use tauri::menu::{CheckMenuItemBuilder, MenuBuilder, MenuItemBuilder};

tauri::Builder::default()
    .setup(|app| {
        let toggle = MenuItemBuilder::with_id("toggle", "Toggle").build(app);
        let check = CheckMenuItemBuilder::new("Mark").build(app);
        let menu = MenuBuilder::new(app).items(&[&toggle, &check]).build()?;

        app.set_menu(menu)?;

        app.on_menu_event(move |app, event| {
            if event.id() == check.id() {
                println!("`check` triggered, do something! is checked? {}", check.is_checked().unwrap());
            } else if event.id() == "toggle" {
                println!("toggle triggered!");
            }
        });
        Ok(())
    })
```

> Notez qu'il y a deux manières de vérifier que "item" de menu est séléctioné: on peut "move" (la fonctionnalités rust des closure) l'"item" vers un gestionnaire d'événement (event handler) sous forme de closure et comparer les IDs, ou d"finir une ID personnalisée via le constructeur `with_id` et utiliser cette ID (chaine de caractère) pour la comparer

Reviewer 2: ```suggestion
Notez qu'il y a deux manières de vérifier qu'un "item" de menu est sélectionné: on peut "move" (la fonctionnalités rust des closure) l'item" vers un gestionnaire d'événement (event handler) sous forme de closure et comparer les IDs, ou d"finir une ID personnalisée via le constructeur `with_id` et utiliser cette ID (chaine de caractère) pour la comparer.
```

> Notez qu'il y a deux manières de vérifier que "item" de menu est séléctioné: on peut "move" (la fonctionnalités rust des closure) l'"item" vers un gestionnaire d'événement (event handler) sous forme de closure et comparer les IDs, ou d"finir une ID personnalisée via le constructeur `with_id` et utiliser cette ID (chaine de caractère) pour la comparer

Reviewer 2: ```suggestion
Notez qu'il y a deux manières de vérifier que "item" de menu est sélectionné: on peut "move" (la fonctionnalités rust des closure) l'item" vers un gestionnaire d'événement (event handler) sous forme de closure et comparer les IDs, ou définir une ID personnalisée via le constructeur `with_id` et utiliser cette ID (chaine de caractère) pour la comparer.
```

Author: On `l'"item"` I'm pretty sure that a single quote can't replace a double, fixed others mistakes

:::tip

> Les "items" de menu peuvent être partagés entre les menus, et l'évenement de menu est lié à un "item" de menu plutôt qu'à un menu ou une "window" 

Reviewer 2: ```suggestion
Les "items" de menu peuvent être partagés entre les menus, et l'événement de menu est lié à un "item" de menu plutôt qu'à un menu ou une "window".
```

> Si vous ne voulez pas que tous les "listeners" soient déclenchés (triggered) lorsqu'un "item" de menu est séléctioné, ne partagez pas les "items" de menu et utilisez des instances dédiés à la place, que vous pouvez "move" dans une closure passée à `tauri::Window/WindowBuilder::on_menu_event`

Reviewer 2: ```suggestion
Si vous ne voulez pas que tous les "listeners" soient déclenchés (triggered) lorsqu'un "item" de menu est sélectionné, ne partagez pas les "items" de menu et utilisez des instances dédiés à la place, que vous pouvez "move" dans une closure passée à `tauri::Window/WindowBuilder::on_menu_event`
```
